pkg/controllers/machinesetsync/machineset_vap_test.go

```diff
@@ -234,4 +235,103 @@ var _ = Describe("MachineSet VAP Tests", func() {
+			Expect(resourceRules).To(HaveLen(1))
+			resourceRules[0].Operations = append(resourceRules[0].Operations, admissionregistrationv1.Update)
+
+			Eventually(k.Update(machineSetVap, func() {
+				admissiontestutils.AddSentinelValidation(machineSetVap)
+				machineSetVap.Spec.MatchConstraints.ResourceRules = resourceRules
+			})).Should(Succeed())
+
+			Eventually(k.Object(machineSetVap), timeout).Should(
+				HaveField("Status.ObservedGeneration", BeNumerically(">=", 2)),
+			)
+
+			By("Updating the VAP binding")
+			policyBinding = &admissionregistrationv1.ValidatingAdmissionPolicyBinding{}
+			Eventually(k8sClient.Get(ctx, client.ObjectKey{
+				Name: "openshift-prevent-mapi-machineset-create-when-capi-exists"}, policyBinding), timeout).Should(Succeed())
+
+			Eventually(k.Update(policyBinding, func() {
+				// paramNamespace=capiNamespace (CAPI resources are params)
+				// targetNamespace=mapiNamespace (MAPI resources are validated)
+				admissiontestutils.UpdateVAPBindingNamespaces(policyBinding, capiNamespace.GetName(), mapiNamespace.GetName())
+			}), timeout).Should(Succeed())
+
+			// Wait until the binding shows the patched values
+			Eventually(k.Object(policyBinding), timeout).Should(
+				SatisfyAll(
+					HaveField("Spec.MatchResources.NamespaceSelector.MatchLabels",
+						HaveKeyWithValue("kubernetes.io/metadata.name",
+							mapiNamespace.GetName())),
+				),
+			)
+
+			By("Creating throwaway MachineSet pair for sentinel validation")
+			mapiMachineSetBuilder = machinev1resourcebuilder.MachineSet().
+				WithNamespace(mapiNamespace.Name)
+
+			sentinelMachineSet := machinev1resourcebuilder.MachineSet().
+				WithNamespace(mapiNamespace.Name).
+				WithName("sentinel-machineset").
+				WithAuthoritativeAPI(mapiv1beta1.MachineAuthorityClusterAPI).
+				Build()
+			Eventually(k8sClient.Create(ctx, sentinelMachineSet), timeout).Should(Succeed())
+
+			capiSentinelMachineSet := clusterv1resourcebuilder.MachineSet().
+				WithName("sentinel-machineset").
+				WithNamespace(capiNamespace.Name).
+				Build()
+			Eventually(k8sClient.Create(ctx, capiSentinelMachineSet)).Should(Succeed())
+
+			Eventually(k.Get(capiSentinelMachineSet)).Should(Succeed())
+
+			admissiontestutils.VerifySentinelValidation(k, sentinelMachineSet, timeout)
+		})
+
+		It("Doesn't allow creation of a MAPI machineset with authoritative API MachineAPI and the same name", func() {
+			By("Create the CAPI MachineSet")
+			Eventually(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+
+			By("Create the MAPI MachineSet")
+			newMapiMachineSet := mapiMachineSetBuilder.
+				WithName("test-machineset").
+				WithAuthoritativeAPI(mapiv1beta1.MachineAuthorityMachineAPI).
+				Build()
+			Eventually(k8sClient.Create(ctx, newMapiMachineSet), timeout).Should(
+				MatchError(ContainSubstring("with authoritativeAPI=MachineAPI because a Cluster API MachineSet with the same name already exists.")))
```

Comment:
```suggestion
				MatchError(ContainSubstring("with spec.authoritativeAPI:MachineAPI because a Cluster API MachineSet with the same name already exists.")))
```